app/templates/partials/applications_table.html

```diff
@@ -2,59 +2,113 @@
 {%- from "govuk_frontend_jinja/components/tag/macro.html" import govukTag -%}
 {%- from "govuk_frontend_jinja/components/table/macro.html" import govukTable -%}
 
-{% macro applications_table(applications, is_past_submission_deadline) %}
+{% macro applications_table(applications, is_past_submission_deadline, show_language_column) %}
 
-{% set ns = namespace(rows = []) %}
-    {% for application in applications %}
-    {% set status_class %}
-    {% if is_past_submission_deadline %}
-        {% if application.status == 'SUBMITTED' %}
-            complete-tag
-        {% elif application.status == 'NOT_SUBMITTED' %}
-            complete-tag
-        {% endif %}
-    {% else %}
-        {%  if application.status == 'SUBMITTED' %}
-            complete-tag
-        {%  elif application.status == 'NOT_STARTED' %}
-            not-started-tag
-        {%  elif application.status == 'IN_PROGRESS' %}
-            in-progress-tag-new
-        {%  elif application.status == 'READY_TO_SUBMIT' %}
-            in-progress-tag-new
-        {% endif %}
-    {% endif %}
-{% endset %}
+	{% set ns = namespace(rows = []) %}
+	{% for application in applications %}
+		{% set status_class %}
+		{% if is_past_submission_deadline %}
+			{% if application.status == 'SUBMITTED' %}
+				complete-tag
+			{% elif application.status == 'NOT_SUBMITTED' %}
+				complete-tag
+			{% endif %}
+		{% else %}
+			{%  if application.status == 'SUBMITTED' %}
+				complete-tag
+			{%  elif application.status == 'NOT_STARTED' %}
+				not-started-tag
+			{%  elif application.status == 'IN_PROGRESS' %}
+				in-progress-tag-new
+			{%  elif application.status == 'READY_TO_SUBMIT' %}
+				in-progress-tag-new
+			{% endif %}
+		{% endif %}
+		{% endset %}
 
-{% if not application.status == 'SUBMITTED' and not is_past_submission_deadline %}
-    {% set application_link = '<a class="govuk-link" href="{}">{}</a>'.format(url_for('application_routes.tasklist', application_id=application.id), gettext('Continue application')) %}
-{% endif %}
+		{% if not application.status == 'SUBMITTED' and not is_past_submission_deadline %}
+			{% set application_link = '<a class="govuk-link" href="{}">{}</a>'.format(url_for('application_routes.tasklist', application_id=application.id), gettext('Continue application')) %}
+		{% endif %}
 
-    {% set row = ns.rows.append(
-    [
-        {
-            'text': application.project_name if application.project_name else gettext('Untitled project')
-        },
-        {
-            'text':  govukTag({
-                'text': application.status|snake_case_to_human ,
-                'classes': status_class
-            })
-        },
-        {
-            'html': application.last_edited|datetime_format_short_month if application.last_edited != None
-        },
-        {
-            'html': application.reference
-        },
-        {
-            'html': application_link
-        }
-    ])%}
-    {% endfor %}
+		{% if show_language_column %}
```

Comment:
As discussed can we try and do this without repeating the whole table?

Reply:
@adamdavies1  How about a Jinja filter?

```python
from jinja2 import Environment

def conditional_list(items):
    return [val for val, hide in items if not hide]

env = Environment()
env.filters['conditional_list'] = conditional_list

# Usage example:
items = [(1, False,), (2, True,), (3, False,),]
template = env.from_string("Filtered items: {{ items|conditional_list }}")
rendered = template.render(items=items)
print(rendered)  # Output: Filtered items: [1, 3]
```

Although it might be a bit annoying to have to specify a tuple for every element - but the function could probably be enhanced to take that into account and default anything else to just returning.